tests/unit/test_dns.py

```diff
@@ -0,0 +1,61 @@
+import pytest
+from app.utils.dns import Service
+
+from collections import Counter
+import dns.message
+
+# DO NOT ADJUST THE FORMATTING ON THIS STRING
+# IT WILL NOT PARSE
+
+SRV_RESPONSE = """id 26729
+opcode QUERY
+rcode NOERROR
+flags QR AA RD RA
+;QUESTION
+nomad.service.city.consul. IN SRV
+;ANSWER
+nomad.service.city.consul. 0 IN SRV 1 70 4646 city1-0114serverless.node.city.consul.
+nomad.service.city.consul. 0 IN SRV 1 15 4647 city2-0114serverless.node.city.consul.
+nomad.service.city.consul. 0 IN SRV 1 15 4647 city3-0114serverless.node.city.consul.
+nomad.service.city.consul. 0 IN SRV 0 15 4647 city4-0114serverless.node.city.consul.
+;AUTHORITY
+;ADDITIONAL
+city1-0114serverless.node.city.consul. 0 IN A 172.31.1.43
+city2-0114serverless.node.city.consul. 0 IN A 172.31.1.167
+city3-0114serverless.node.city.consul. 0 IN A 172.31.1.76
+city4-0114serverless.node.city.consul. 0 IN A 172.31.1.55"""
+
+
+@pytest.fixture
+def response():
+    return dns.message.from_text(SRV_RESPONSE)
+
+
+class TestServiceDiscovery(object):
+    def test_service_inits_with_srv_result(self, response):
+        """ Service Entries are inited with Service Result"""
+        srv = Service(response)
+        assert srv.url in ["172.31.1.43:4646", "172.31.1.167:4647", "172.31.1.76:4647"]
```

Comment:
I suggest that the IPs and ports be labeled in variables so that there's meaning attached to them.  Otherwise it looks like magic numbers

Reply:
These are magic numbers

Reply:
They come from SRV response so they wont be changing